### ISSUE TITLE
TestClient.request() method does not have a data_type argument like the real client

### ISSUE THREAD

**Issue by roblight** [lwinterface/panini#239]
Here's what I quickly came up with:

https://github.com/lwinterface/panini/compare/master...roblight:panini:master

The following tests are failing and I have not looked into why yet:

```
tests/test_diff_datatypes.py::test_listen_bytes FAILED  [  6%]
tests/test_diff_datatypes.py::test_publish_str FAILED   [  8%]
tests/test_diff_datatypes.py::test_publish_bytes FAILED [  9%]
tests/test_diff_datatypes.py::test_request_str FAILED   [ 12%]
tests/test_diff_datatypes.py::test_request_bytes FAILED [ 13%]
```
It would be great if someone with more knowledge of this project could add this argument! :smile:

**Comment by artas728** [comment]
Indeed, need to be fixed. We will think about the best way to do this.

**Comment by artas728** [comment]
Same as for #238. It's fixed for v0.8.0a1. It'll take some time before the release of v0.8.0.

But looking ahead, I'm not sure we'll continue to support an additional test client for a long. We are thinking about a way to use Panini itself as a test client for integration tests.
